### ISSUE TITLE
[Improvement] Extract email validation for common usage

### ISSUE THREAD

**Issue by hoangnguyen92dn** [thiennguyen0196/survey#16]
## Issue

There is the same logic to validate an email in `ForgotViewModel` and `LoginViewModel` so I'd suggest extracting this to a function for common usage.

https://github.com/thiennguyen0196/survey/blob/2f0fc36adcb5fd30fd78ef3f394698f860939af6/app/src/main/java/com/thiennguyen/survey/feature/forgot/ForgotViewModel.kt#L24-L28

https://github.com/thiennguyen0196/survey/blob/2f0fc36adcb5fd30fd78ef3f394698f860939af6/app/src/main/java/com/thiennguyen/survey/feature/login/LoginViewModel.kt#L25-L29

## Expected

The email validation function will be created and invoked in the ViewModel classes.

**Comment by thiennguyen0196** [comment]
This is fixed in PR #22

**Comment by hoangnguyen92dn** [comment]
@thiennguyen0196 For this issue, I am not sure about your idea by using `Inline class` rather than extension 🤔 Can you please explain in detail why you would go with the current approach instead?

**Comment by thiennguyen0196** [comment]
@hoangnguyen92dn 
Indeed using extension was my first thought to extract the common usage function. However, in this case, I also want to distinguish between email text and password text so they cannot be passed wrongly in the arguments of `viewModel.submitLogin()` function. For example, if they are still string type, there is a chance that we could pass password text for email field and vice versa. 

To sum up, I use inline class to have more strict usage in passing arguments with `viewModel.submitLogin()` function 😄

**Comment by hoangnguyen92dn** [comment]
Got your idea now. Thanks for your explanation 👍
